### PR TITLE
【Comm】Enable FLAGS_dynamic_static_unified_comm in legacy ut

### PR DESCRIPTION
**Pull request by Xing-lil**
### PR Category

Communication Library

### PR Types

Improvements

### Description

1. Switch legacy FLAGS_dynamic_static_unified_comm to true.
2. Unify `get_model` and `get_model_new_comm` in comm ut.

NEXT

1. Fix barrier and sendrecv op with FLAGS_dynamic_static_unified_comm.
2. Delete the redundant code.
https://github.com/PaddlePaddle/Paddle/pull/67005

Pcard-70448

### CODE REVIEW

File: test/collective/column_parallel_linear_api.py [hunk 1]
```diff
@@ -60,6 +60,39 @@ def get_model(self, main_prog, startup_program, rank):
 
             return [linear_out]
 
+    def get_model_new_comm(
```
Comment (liym27) [inline test/collective/column_parallel_linear_api.py]: 已线下沟通。
建议不在每个单测文件里单独增加 get_model_new_comm，代码冗余；直接在 test_collective_api_base.py 里新通信库复用 get_model 即可。本次PR后，单测将都跑新通信库，不再跑旧通信库。

File: test/distributed_passes/auto_parallel_pass_test_base.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ class AutoParallelPassTestBase(DistPassTestBase):
     def setUp(self):
         paddle.enable_static()
         seed = int(os.environ.get('SEED', -1))
-        os.environ["FLAGS_dynamic_static_unified_comm"] = "0"
+        os.environ["FLAGS_dynamic_static_unified_comm"] = "1"
```
Comment (zhiqiu) [inline test/distributed_passes/auto_parallel_pass_test_base.py]: =1 is no need since it is true by default

Comment (Xing-lil) [reply test/distributed_passes/auto_parallel_pass_test_base.py]: I will delete the redundant code. Thx

**Review by liym27 — APPROVED** [review]
LGTM

File: test/collective/column_parallel_linear_api.py (hunk 1 above)
Comment (Xing-lil) [reply test/collective/column_parallel_linear_api.py]: 已修改，整合了get_model_new_comm和get_model的代码，并直接在test_collective_api_base.py复用。感谢！

**Review by zhiqiu — COMMENTED** [review]
lgtm

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**